RecoPixelVertexing/PixelTrackFitting/interface/RZLine.h

```diff
@@ -0,0 +1,120 @@
+#ifndef PixelTrackFitting_RZLine_H
+#define PixelTrackFitting_RZLine_H
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/GlobalError.h"
+#include "CommonTools/Utils/interface/DynArray.h"
+
+#include "CommonTools/Statistics/interface/LinearFit.h"
+
+#include <vector>
+
+class RZLine {
+public:
+  struct ErrZ2_tag {};
+
+  /**
+   * Constructor for containers of GlobalPoint, GlobalError, and bool
+   *
+   * @tparam P  Container of GlobalPoint
+   * @tparam E  Container of GlobalError
+   * @tparam B  Container of bool
+   *
+   * Container can be e.g. std::vector, std::array, or DynArray.
+   *
+   * Although for std::array use this constructor could be specialized
+   * to use std::array instead of DynArray for temporary storage.
+   */
+  template <typename P, typename E, typename B>
+  RZLine(const P& points, const E& errors, const B& isBarrel) {
+    const size_t n = points.size();
+    declareDynArray(float, n, r);
+    declareDynArray(float, n, z);
+    declareDynArray(float, n, errZ2);
+    for(size_t i=0; i<n; ++i) {
+      const GlobalPoint& p = points[i];
+      r[i] = p.perp();
+      z[i] = p.z();
+    }
+
+    float simpleCot2 = sqr( (z[n-1]-z[0])/ (r[n-1]-r[0]) );
```

Comment:
(Note that this code is exactly as before) I guess it could, in principle, happen that the first and last hit would have exactly the same `r`, if all hits come from FPix/TID/TEC. But I'd expect this situation to be very unlikely, especially given the constraints of seeds to point towards beamspot (even if loosely on strip-triplet steps).